templates/test/ci/cluster-template-prow-external-cloud-provider-ci-version.yaml

```diff
@@ -76,7 +76,8 @@ spec:
           cloud-config: /etc/kubernetes/azure.json
           cloud-provider: external
           cluster-name: ${CLUSTER_NAME}
-          feature-gates: HPAContainerMetrics=true
+          external-cloud-volume-plugin: azure
```

Comment:
@jackfrancis -- Any idea why this feature gate is not coming into the generated files. I did not remove `HPAContainerMetrics=true` from anywhere and I can see that it is still present in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/patches/controller-manager-featuregates.yaml#L10

Reply:
Also, what is the use of this HPAMetric feature gate flag? Is it required in the CI?

Reply:
This test template is curated in part by the folks who maintain cloud-provider-azure. I assume this feature gate enables specific cloud-provider-azure E2E tests (they use capz to build clusters to run their E2E tests against).

So tl;dr we can't remove it from this template.

cc @lzhecheng @nilo19

Reply:
@sonasingh46 try adding the new CSI feature gate to this file:

templates/test/ci/patches/controller-manager-featuregates.yaml

E.g.: `feature-gates: HPAContainerMetrics=true,CSIMigrationAzureDisk=true`

and remove from the `controllerManager` part of this file:

`templates/flavors/external-cloud-provider/patches/external-cloud-provider.yaml`

then re-run `hack/gen-flavors.sh`

Reply:
We can remove this because for now there is no horizontal pod autoscaler in our cluster, so this feature gate isn't needed.

Reply:
If we want to update the HPA feature gates let's do that as a separate PR so @sonasingh46 doesn't have to worry about it. Also, it will be easier to revert that change if we need to if it's in its own PR.